pkg/volume/volume.go

```diff
@@ -0,0 +1,53 @@
+package volume
+
+import (
+	"context"
+	"fmt"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	kclient "github.com/acorn-io/acorn/pkg/k8sclient"
+	"golang.org/x/exp/slices"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetVolumeClassNames(ctx context.Context, c client.Client) ([]string, error) {
+	volumeClasses := new(apiv1.VolumeClassList)
+	if err := c.List(ctx, volumeClasses, &kclient.ListOptions{}); err != nil {
+		return nil, err
+	}
+
+	volumeClassNames := make([]string, 0, len(volumeClasses.Items)+1)
+	// Any volume can be ephemeral.
+	volumeClassNames = append(volumeClassNames, v1.VolumeRequestTypeEphemeral)
+	for _, sc := range volumeClasses.Items {
+		volumeClassNames = append(volumeClassNames, sc.Name)
+	}
+
+	return volumeClassNames, nil
+}
+
+func CheckVolumeClasses(ctx context.Context, c client.Client, appInstanceSpec v1.AppInstanceSpec, appSpec *v1.AppSpec) *field.Error {
+	if len(appInstanceSpec.Volumes) == 0 && len(appSpec.Volumes) == 0 {
```

Comment:
the appinstancespec and appspec should always be consistent, there won't be a scenario where appInstanceSpec =1 volume and appSpec = 0 right?

Reply:
that makes sense why it is both have to be = 0 since there are two ways to define volumes, thank you